### PR TITLE
add retries on repoquery

### PR DESCRIPTION
**Pull request by sosiouxme**
Should address bug https://bugzilla.redhat.com/show_bug.cgi?id=1482551

Extracted these changes from https://github.com/openshift/openshift-ansible/pull/5125 since that is parked at the moment and the repoquery changes are independent of the yum ones.

### CODE REVIEW

**Comment by sosiouxme** [comment]
aos-ci-test

File: roles/docker/tasks/package_docker.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@
   command: "{{ repoquery_cmd }} --installed --qf '%{version}' docker"
   when: not openshift.common.is_atomic | bool
   register: curr_docker_version
+  retries: 4
```
Comment (ashcrow) [inline roles/docker/tasks/package_docker.yml]: Should this be modified to use the ``repoquery`` module with the updates in this PR?

Comment (sosiouxme) [reply roles/docker/tasks/package_docker.yml]: Probably. I think it's the only time this is used. I didn't want to take the time to figure out if there was any gotcha to doing that.

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by sosiouxme** [comment]
[test]

**Comment by ashcrow** [comment]
```
[Conformance][networking][router] openshift router metrics
/go/src/github.com/openshift/origin/test/extended/router/metrics.go:192
  The HAProxy router
  /go/src/github.com/openshift/origin/test/extended/router/metrics.go:191
    should expose prometheus metrics for a route [It]
    /go/src/github.com/openshift/origin/test/extended/router/metrics.go:173

    Expected error:
        <*errors.errorString | 0xc4219ebef0>: {
            s: "last response from server was not 403: 401",
        }
        last response from server was not 403: 401
    not to have occurred
```

**Comment by ashcrow** [comment]
[test]

**Comment by sdodson** [comment]
[test]

**Comment by sosiouxme** [comment]
Looks like https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible/671/ hit every common flake in the book. Added these that I've seen but not reported:

- https://github.com/openshift/origin/issues/16368
- https://github.com/openshift/origin/issues/16369

This PR is a change to retry failed installer queries for packages. I didn't see any installer failures. It seems unlikely any failures outside install/upgrade could be a result of this change. I'd like it to get in a build in the morning if there is one, so I'm gonna merge it.

**Pull request merged**